odoo_module_migrate/migration.py

```diff
@@ -91,9 +92,25 @@ def __init__(
         for module_name in module_names:
             self._module_migrations.append(ModuleMigration(self, module_name))
 
+        if os.path.exists(".pre-commit-config.yaml") and self._pre_commit:
+            self._run_pre_commit(module_names)
+
         # get migration scripts, depending to the migration list
         self._get_migration_scripts()
 
+    def _run_pre_commit(self, module_names):
+        logger.info("Run pre-commit")
+        _execute_shell(
+            "pre-commit run -a", path=self._directory_path, raise_error=False)
+        if self._commit_enabled:
+            logger.info("Stage and commit changes done by pre-commit")
+            _execute_shell(
+                "git commit -a -m '[IMP] %s: black, isort, prettier' --no-verify"
```

Comment:
You have to add a `git add -A` command before. `git commit -a` only stages modified files, not added ones, and we have some hooks that add the `setup` boilerplate.

Also, I'm more biased about changing that message to `[IMP] %s: pre-commit execution`. Apart from black, isort and prettier, we have autoflake8, eslint, and possibly other hooks that modify stuff. It's not fair to exclude them. 😄

Reply:
Oh right, I restored `git add -A` and also changed the commit message :+1: